servant-server/src/Servant/Server/Internal.hs

```diff
@@ -405,6 +406,48 @@ instance (KnownSymbol sym, HasServer api context)
           examine v | v == "true" || v == "1" || v == "" = True
                     | otherwise = False
 
+-- | If you use @'QueryParamForm' BookSearchParams@ in one of the endpoints for your API,
+-- this automatically requires your server-side handler to be a function
+-- that takes an argument of type @['BookSearchParams']@.
+--
+-- This lets servant worry about all key-values in the query string
+-- and turning each of them into a value of the type you specify.
+--
+-- You can control how the individual values are converted from 'BookSearchParams'
+-- to your type by simply providing an instance of 'FromForm' for your type.
+--
+-- Example:
+--
+-- > data BookSearchParams = BookSearchParams
+-- >   { title :: Text
+-- >   { authors :: [Text]
+-- >   , page :: Maybe Int
+-- >   } deriving (Generic)
+-- > instance FromForm BookSearchParams
```

Comment:
We should probably mention that we default implementation uses `FromHttpApiData` for each field.